cpp/src/io/csv/csv_common.h

```diff
@@ -35,6 +38,13 @@ enum : uint8_t {
 };
 using flags = uint8_t;
 
+struct column_builder {
```

Comment:
is this struct expected to take over (some of the) creation logic, or just group these pieces of information?

Reply:
`column_builder` is a misnomer.  For now, I'm grouping parameters and trying to eliminate the idea of "active" vs "actual" rows, which should help reduce the parameter count significant. It's more like `column_being_built`.

Reply:
`column_builder` is essentially used to replace column out params (data, mask, null count) with a single out param. This will hopefully go away as we begin returning data from these functions rather than using out params. In the meantime, however, it allows the refactoring to happen piece-wise, rather than needing the entire decoding process to be updated all at once.

Ideally (for csv, as that's what we're discussing here), we'd have a workflow like this, passing previous step returned results in to the following steps.

0) pre-process source data (decode, etc)
1) detect/generate the column names and types, returning an appropriate type.
2) filter column/header configs based on the user's desired column, returning an appropriate type.
3) decode the rows, returning the columns (data, mask, null count).
4) post-process columns (string gather/replace, maybe as a single kernel, depending on code reuse).
5) generate and return table + metadata 

Each of these steps would individually be responsible for creating output for each column (even if that output is just a placeholder, thereby eliminating the need for the idea of "active" columns and separate length vectors/spans (which reduces arg count, if the number of columns is somehow big enough to require too much memory, we can possible use a map which points to the "active" column data with offsets).

Reply:
We do have `column_buffer`. Maybe it can be modified to serve as `column_builder`.

Reply:
Maybe, but `column_buffer` is used in multiple places. `column_builder` would be namespace'd to csv.

Reply:
The suggestion is to use `column_buffer` if the parameters together indeed form a proto-column. When there are more parameters to be grouped, then yes, we most likely need separate `xyz_builder` types in different components.